src/app/auth/auth.guard.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, CanActivate, CanDeactivate, Router, RouterStateSnapshot, UrlTree } from '@angular/router';
+import { Observable } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthGuard implements CanActivate, CanDeactivate<unknown> {
+
+  constructor(
+    private router: Router
+  ){}
+
+  canActivate(
+    route: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+      if(localStorage.getItem('token') || sessionStorage.getItem('token')){
```

Comment:
Please move 'token' to consts since it's used in login component as well

Reply:
done